trtools/utils/utils.py

```diff
@@ -6,73 +6,70 @@
 import itertools
 import math
 import os
-import sys
-from typing import Any, Dict
+from typing import Any, Dict, List, Optional
 
+import cyvcf2
 import numpy as np
 import scipy.stats
-import vcf
 
 import trtools.utils.common as common # pragma: no cover
 
 nucToNumber={"A":0,"C":1,"G":2,"T":3}
-    
-def LoadSingleReader(vcffile, checkgz=True, region=None):
-    r"""Return VCF reader
+
+def LoadSingleReader(
+        vcf_loc: str,
+        checkgz: bool = True) -> Optional[cyvcf2.VCF]:
+    """
+    Return a VCF reader
 
     Parameters
     ----------
-    vcffile : str
-        VCF files to read
-    checkgz: boolean, optional
-        Check whether VCF file is gzipped and indexed
-    region : str, optional
-        Chrom:start-end to restrict to
-    
+    vcf_loc :
+        The location of the VCF file to read
+    checkgz:
+        Check whether VCF file is gzipped and indexed,
+        if not return None
+
     Returns
     -------
-    reader : vcf.Reader
-        VCF reader
+    reader : Optional[cyvcf2.VCF]
+        The cyvcf2.VCF instance, or None if the VCF is not present
     """
-    if not os.path.isfile(vcffile):
-        common.WARNING("Could not find VCF file %s"%vcffile)
+    if not os.path.isfile(vcf_loc):
+        common.WARNING("Could not find VCF file %s"%vcf_loc)
         return None
     if checkgz:
-        if not vcffile.endswith(".vcf.gz") and not vcffile.endswith(".vcf.bgz"):
-            common.WARNING("Make sure %s is bgzipped and indexed"%vcffile)
+        if not vcf_loc.endswith(".vcf.gz") and not vcf_loc.endswith(".vcf.bgz"):
+            common.WARNING("Make sure %s is bgzipped and indexed"%vcf_loc)
             return None
-        if not os.path.isfile(vcffile+".tbi"):
-            common.WARNING("Could not find VCF index %s.tbi"%vcffile)
+        if not os.path.isfile(vcf_loc+".tbi"):
+            common.WARNING("Could not find VCF index %s.tbi"%vcf_loc)
             return None
-    if vcffile.endswith(".vcf.gz") or vcffile.endswith(".vcf.bgz"):
-        reader = vcf.Reader(open(vcffile, "rb"))
-    else:
-        reader = vcf.Reader(open(vcffile))
-    
-    if region is None:
-        return reader
-    else: return reader.fetch(region)
-
-def LoadReaders(vcffiles, checkgz=True, region=None):
-    r"""Return list of VCF readers
+    return cyvcf2.VCF(vcf_loc)
+
+def LoadReaders(
+        vcf_locs: List[str],
+        checkgz: bool = True) -> Optional[List[cyvcf2.VCF]]:
+    """
+    Return a list of VCF readers
 
     Parameters
     ----------
-    vcffiles : list of str
-        List of VCF files to merge
-    checkgz: bool, optional
+    vcf_locs :
+        A list of vcf locations
+    checkgz:
         Check if each VCF file is gzipped and indexed
-    region : str, optional
-        Chrom:start-end to restrict to
-    
+        If any are not, return None
```

Comment:
Does this mean that if a file is not gzipped, checkgz is going to be None?
I think the name and the functionality of the variable don't match really well. But I'm guessing this is not a recent change, so we can fix it after this round of updates.

Reply:
It means that if checkgz is True (the default), then if any of the VCFs are not gzipped, none of them are returned. (None is returned instead). Otherwise, a list of cyvcf2.VCF objects is returned.

If it is False, then a list of VCFs will be returned if all of them can be found, regardless of if some were or were not gzipped.

Happy to clean this up at the end.